src/Storages/Kafka/ReadBufferFromKafkaConsumer.cpp

```diff
@@ -82,18 +82,27 @@ ReadBufferFromKafkaConsumer::ReadBufferFromKafkaConsumer(
 
 ReadBufferFromKafkaConsumer::~ReadBufferFromKafkaConsumer()
 {
-    /// NOTE: see https://github.com/edenhill/librdkafka/issues/2077
     try
     {
         if (!consumer->get_subscription().empty())
             consumer->unsubscribe();
-        if (!assignment.empty())
-            consumer->unassign();
-        while (consumer->get_consumer_queue().next_event(100ms));
     }
     catch (const cppkafka::HandleException & e)
     {
-        LOG_ERROR(log, "Exception from ReadBufferFromKafkaConsumer destructor: " << e.what());
+        LOG_ERROR(log, "Exception from ReadBufferFromKafkaConsumer destructor (unsubscribe): " << e.what());
+    }
+
+    try
+    {
+        // we need to drain rest of the messages / queued callback calls from the consumer
+        // after unsubscribe, otherwise consumer will hang on destruction
+        // see https://github.com/edenhill/librdkafka/issues/2077
+        //     https://github.com/confluentinc/confluent-kafka-go/issues/189 etc.
+        while (consumer->poll(100ms));
```

Comment:
but if autocommit will be enabled (yes it turned off by default, but still) then it could potentially steal some messages? maybe autocommit should be disabled/warned (`StorageKafka::updateConfiguration`)?

Reply:
1) AFAIK you can commit (also with autocommit feature) into consumer after unsubscribe 
2) autocommit should never be used - it is a bad idea and the code is not ready for that in general,

Reply:
I'll make a fast fix for that.